dotCMS/src/main/java/com/dotcms/ema/EMAWebInterceptor.java

```diff
@@ -0,0 +1,208 @@
+                final ProxyResponse pResponse = proxy.sendPost(proxyUrl.get(), params);
+
+                if (pResponse.getResponseCode() == 200) {
+                    responseStr = new String(pResponse.getResponse());
+                }else {
+                    responseStr+="<html><body>";
+                    responseStr+="<h3>Unable to connect with the rendering engine</h3>";
+                    responseStr+="<br><div style='display:inline-block;width:80px'>Trying: </div><b>" + proxyUrl.get()  + "</b>";
+                    responseStr+="<br><div style='display:inline-block;width:80px'>Got:</div><b>" + pResponse.getStatus() + "</b>";
+                    responseStr+="<hr>";
+                    responseStr+="<h4>Headers</h4>";
+                    responseStr+="<table border=1 style='min-width:500px'>";
+
+                    for(Header header : pResponse.getHeaders()) {
+                      responseStr+="<tr><td style='font-weight:bold;padding:5px;'><pre>" + header.getName() + "</pre></td><td><pre>" + header.getValue() + "</td></tr>";
+                    }
+                    responseStr+="</table>";
+                    
+                    responseStr+="<p>The Json Payload, POSTing as Content-Type:'application/x-www-form-urlencoded' with form param <b>dotPageData</b>, has been printed in the logs.</p>";
+                    responseStr+="</body></html>";
+
+                }
+
+                json.getJSONObject("entity").getJSONObject("page").put("rendered", responseStr);
+                json.getJSONObject("entity").getJSONObject("page").put("remoteRendered", true);
+                response.setContentType("application/json");
+
+                response.getWriter().write(json.toString());
+            }
+        } catch (Exception e) {
+            Logger.error(this, e.getMessage());
+        }
+
+        return true;
+    }
+
+
+    /**
+     * Gets the proxyUrl for the given host
+     * @param currentHost current Host to get the proxyUrl value
+     * @return Optional String of the proxyUrl, if there is not found or an error is thrown returns an empty
+     */
+    private Optional<String> proxyUrl(final Host currentHost) {
+        AppSecrets appSecrets = null;
+
+        try{
+            appSecrets = APILocator.getAppsAPI().getSecrets(EMA_APP_CONFIG_KEY,
+                    true, currentHost, APILocator.systemUser()).get();
+            return appSecrets.getSecrets().containsKey(PROXY_EDIT_MODE_URL_VAR)?
+                    Optional.ofNullable(appSecrets.getSecrets().get(PROXY_EDIT_MODE_URL_VAR).getString()) : Optional.empty();
+
+        } catch (DotSecurityException | DotDataException e) {
+            Logger.error(this, e.getMessage());
+            return Optional.empty();
+        } finally {
+            if(UtilMethods.isSet(appSecrets)){
+                appSecrets.destroy();
+            }
+        }
+    }
+
+
+    @CloseDBIfOpened
+    private Host getCurrentHost(HttpServletRequest request) {
```

Comment:
Can't we just use `com.dotmarketing.business.web.HostWebAPI#getCurrentHost(javax.servlet.http.HttpServletRequest, com.liferay.portal.model.User)` ?

Reply:
done